### DISCUSSION TITLE
BIRT 4.8 end-of-life

### DISCUSSION THREAD

**Discussion by SteveSchafer-Innovent** [eclipse-birt/birt#912] [General]
One of our clients has asked when BIRT 4.8 will reach end-of-life.  Is end-of-life even a concept relevant to BIRT?

**Comment by ruspl-afed** [comment]
The Eclipse Project in the whole could be scheduled to termination [1] 

For each particular released version it is up to Project team to decide should it invest to further support or not. And if Project team has no resources to support some particular version it could be a sponsored activity. Technically it is possible.

[1] https://www.eclipse.org/projects/handbook/#starting-project-phases

**Comment by hvbtup** [comment]
I can only speak for myself. My personal opinion is:

We are only a handful of people who are maintaining BIRT (including you) - some (or most?) of them doing this as a side project - at least in my case.
BIRT is _used_ by many (including some big players), but - as always in the OS world - only very few are _contributing_.
The situation is is in a way comparable to Log4J - we are more than one maintainer of course, but the software is much more complex, too.

We just don't have the resources to fix bugs in older releases unless the team increases in number.

This is open source, and now, since _building from source_ is working well, if a bug in older release annoys you (meaning anyone from here on), you can always choose to

a) live with it
b) update to the newest release (if the bug is fixed here already)
c) use the latest build artifacts from GH (if the bug is fixed here already)
d) try to "patch" the old release by only compiling part of the sources including the bug fix (if the bug is fixed in the GH sources already).
e) fix the bug yourself in the old release
f) fix the bug yourself in sources from the GH master.

In case of e) and f) you must make the source code available under the EPL-2.0 if you distribute code or binaries built from that code outside your company. If you only use it internally, you should nevertheless contribute the bugfix as a good practice - this makes life easier for yourself and for others.

As Alexander mentioned, you can also try to pay someone else for helping.

Note: I don't say that nobody is willing to help in _analyzing_ issues with older releases.
But we just can't afford wasting our scarce resources with fixing bugs in old releases.

If the rate of new issues should increase in the future, but not the number of persons wo actively help, I'd even go one step further and tell anyone to reproduce issues with the latest release or a fresh build artifact from GH before posting a new issue.
